seisbench/models/eqtp.py

```diff
@@ -0,0 +1,306 @@
+import warnings
+from typing import Any
+
+import numpy as np
+import torch
+import torch.nn as nn
+
+import seisbench.util as sbu
+from .eqtransformer import EQTransformer, SeqSelfAttention, Decoder
+
+
+class EQTP(EQTransformer):
+    """
+    The EQTP from Peng et al. (2025)
+
+    It is an extended version of the EQTransformer model, which builds upon its
+    phase picking capabilities by adding P-wave polarity determination functionality.
+
+    This model is designed for processing three-component seismic waveform data,
+    and can simultaneously output picking results for phases such as P-waves/S-waves
+    and polarity determination results for P-waves (Up U/Down D/Unknown N).
+
+    Implementation is adapted from the EQTransformer with SeisBench GitHub repository (https://github.com/seisbench/seisbench).
+
+    the EQTP model can be instantiated via the `from_pretrained("NCEDC")` method.
+
+    .. document_args:: seisbench.models EQTP
+
+    """
+
+    _annotate_args = EQTransformer._annotate_args.copy()
+    _annotate_args["polarity_threshold"] = ("Polarity threshold", 0.3)
+    _annotate_args["polarity_dict"] = ("The position of the polarity vector represents the polarity", {
+        2: "N",
+        0: "U",
+        1: "D"
+    })
+
+    def __init__(
+            self,
+            in_channels=3,
```

Comment:
Can you delete all constructor parameters that aren't explicitly used (like `in_channels` and `in_samples`), so that they are just passed through using kwargs?